smsdk/_version.py

```diff
@@ -0,0 +1,82 @@
+#
+"""
+Track version information for factorytx-core module here.
+"""
+import typing
+import itertools
+import requests
+import warnings
+import functools
+
+
+class VersionInfo(typing.NamedTuple):
+    major: typing.Any
+    minor: typing.Any
+    patchlevel: typing.Any
+    releaselevel: typing.Any
+    serial: typing.Any
+
+    def __str__(self) -> str:
+        result = "".join(
+            itertools.chain(
+                ".".join(str(i) for i in self[:3] if i is not None),
+                (str(i) for i in self[3:] if i is not None),
+            )
+        )
+        return result
+
+
+version_info = VersionInfo(1, 26, 0, "", None)
+version = str(f"v{version_info}")
+
+
+def get_latest_sdk_release() -> typing.Optional[typing.Any]:
+    api_url = f"https://api.github.com/repos/{owner}/{repo}/releases"
+
+    try:
+        response = requests.get(api_url)
+
+        # To raise an HTTPError for any bad responses
+        response.raise_for_status()
+
+        releases = response.json()
+        if releases:
+            latest_release = releases[0]
+            return latest_release["tag_name"]
```

Comment:
I think this will get you into trouble. Better to sort the identifiers yourself so you know the order.  Github could change the order of releases, or there could be a bug fix for an older release.

Reply:
Implemented sorting of GitHub releases to obtain the latest version.